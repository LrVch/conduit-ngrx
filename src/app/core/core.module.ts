import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
