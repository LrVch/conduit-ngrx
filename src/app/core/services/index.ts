export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
