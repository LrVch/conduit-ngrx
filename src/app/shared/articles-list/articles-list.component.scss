.articles-list__loader {
    height: 20px;
}