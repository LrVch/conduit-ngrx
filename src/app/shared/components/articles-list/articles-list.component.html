<div class="articles-list">
  <div class="articles-list__preview"
       [hidden]="isLoading || articlesList.length">
    {{ 'conduit.articlesList.empty' | translate }}.
  </div>

  <div>
    <app-article [article]="article"
                 [locale]="locale"
                 (favorited)="onFavoriteTooggle($event)"
                 *ngFor="let article of articlesList; trackBy: trackArticle"></app-article>

    <div *ngIf="!articlesList.length && isLoading">
      <app-dummy-article-item *ngFor="let d of dummys$ | async; trackBy: trackDummy"></app-dummy-article-item>
    </div>
  </div>
</div>