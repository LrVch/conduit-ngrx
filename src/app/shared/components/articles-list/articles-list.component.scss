.articles-list__loader {
    height: 5px;
}