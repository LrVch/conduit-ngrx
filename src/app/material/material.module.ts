import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
];

@NgModule({
  imports: [...modules],
  exports: [...modules]
})
export class MaterialModule { }
