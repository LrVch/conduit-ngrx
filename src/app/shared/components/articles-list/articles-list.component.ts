import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Article } from '@app/core';
import { articleListItemAnimation, articleListAnimation } from '@app/shared/animations';
import { of, range } from 'rxjs';
import { toArray } from 'rxjs/operators';

@Component({
  selector: 'app-articles-list',
  templateUrl: './articles-list.component.html',
  styleUrls: ['./articles-list.component.scss'],
  animations: [
    articleListItemAnimation,
    articleListAnimation
  ]
})
export class ArticlesListComponent {
  @Input('locale') locale: string;
  @Input('articlesList') articlesList: Article[] = [];
  @Input('isLoading') isLoading = false;
  @Input('isErrorLoading') isErrorLoading = false;
  @Output() favorited = new EventEmitter<Article>();
  @Input('contentLength') contentLength = 0;

  onFavoriteTooggle(article: Article): void {
    this.favorited.emit(article);
  }

  get dummys$() {
    return range(0, this.contentLength).pipe(toArray());
  }
}
