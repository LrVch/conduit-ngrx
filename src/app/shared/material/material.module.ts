import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
