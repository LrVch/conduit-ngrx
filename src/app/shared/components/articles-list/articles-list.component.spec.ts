import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '@app/lib/testing';

import { ArticlesListComponent } from './articles-list.component';
import { DebugElement, Component, Input, Output, EventEmitter } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';
import { Article } from '@app/core';
import { getArticles } from '@app/lib/testing/mock-data.helpers';

@Component({
    selector: 'app-article',
    template: `
    {{ article.title }}
    <a (click)="onFavoriveToggle()"></a>
    `
})
class ArticleComponent {
    @Input('article') article: Article = {} as Article;
    @Output() favorited = new EventEmitter<Article>();

    onFavoriveToggle(): void {
        this.favorited.emit(this.article);
    }
    constructor() { }
}

describe('ArticlesListComponent', () => {
    let component: ArticlesListComponent;
    let fixture: ComponentFixture<ArticlesListComponent>;
    let de: DebugElement;
    let el: HTMLElement;
    const event = {
        stopPropagation() { }
    };
    let articles: Article[];

    beforeEach(
        async(() => {
            articles = getArticles(3);
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [ArticlesListComponent, ArticleComponent],
                    imports: [MaterialModule]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(ArticlesListComponent);
                component = fixture.componentInstance;
                component.articlesList = articles;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has "isLoading" @Input', () => {
        expect(component.isLoading).toBe(false);
    });

    it('should has "isErrorLoading" @Input', () => {
        expect(component.isErrorLoading).toBe(false);
    });

    it('should has "articlesList" @Input', () => {
        expect(component.articlesList).toEqual(articles);
    });

    it('should show "articlesList" @Input', () => {
        component.articlesList = getArticles(3);
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();

        component.articlesList = getArticles(2);
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();

        component.articlesList = getArticles(1);
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();

        component.articlesList = [];
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should show message "No articles are here... yet"', () => {
        component.articlesList = [];
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();

        component.isLoading = false;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();

    });

    it('should show loader', () => {
        component.isLoading = true;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should raise favorited event when clicked (selecteTag) and select first article', () => {
        let selectedArticle: Article;
        const arts = getArticles(3);
        arts[0].title = 'selected title';
        component.articlesList = arts;
        component.favorited.subscribe((a: Article) => selectedArticle = a);

        fixture.detectChanges();

        const firstTag = de.query(By.css('app-article')).query(By.css('a'));

        firstTag.triggerEventHandler('click', event);
        expect(selectedArticle).toBe(arts[0]);
    });
});

