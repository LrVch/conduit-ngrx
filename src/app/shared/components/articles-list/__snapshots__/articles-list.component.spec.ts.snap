// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesListComponent should show "articlesList" @Input 1`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
      hidden=""
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show "articlesList" @Input 2`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
      hidden=""
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show "articlesList" @Input 3`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
      hidden=""
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show "articlesList" @Input 4`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show loader 1`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading={[Function Boolean]}
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
      hidden=""
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      <app-article
        ng-reflect-article="[object Object]"
      >
         title 
        <a />
      </app-article>
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show message "No articles are here... yet" 1`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      
    </div>
  </div>
</app-articles-list>
`;

exports[`ArticlesListComponent should show message "No articles are here... yet" 2`] = `
<app-articles-list
  articlesList={[Function Array]}
  contentLength="0"
  deleteArticle={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  isErrorLoading="false"
  isLoading="false"
>
  <div
    class="articles-list"
  >
    <div
      class="articles-list__preview"
    >
       conduit.articlesList.empty. 
    </div>
    <div>
      
      
    </div>
  </div>
</app-articles-list>
`;
